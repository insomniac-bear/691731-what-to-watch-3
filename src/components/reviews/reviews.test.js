import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import NameSpace from '../../reducer/name-space.js';
import Reviews from './reviews.jsx';

const mockStore = configureStore([]);

const mockData = [
  {
    id: 0,
    user: {
      id: 2,
      name: `Author name`,
    },
    rating: 9.9,
    comment: `Some comment`,
    date: `January 01, 2001`,
  }
];

it(`Render Reviews tab`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      commentsToFilm: mockData,
    },
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <Reviews
            comments={mockData}
          />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
