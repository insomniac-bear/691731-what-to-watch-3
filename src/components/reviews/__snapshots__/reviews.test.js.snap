// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Reviews tab 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Some comment
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Author name
          </cite>
          <time
            className="review__date"
            dateTime="January 01, 2001"
          >
            January 01, 2001
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        9.9
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  />
</div>
`;
